### DISCUSSION TITLE
Installing Xcode via Cask?

### DISCUSSION THREAD

**Discussion by elsiehupp** [Homebrew/discussions#845] [Casks]
This might be a dumb question, but considering it's possible to download and install Xcode via a `.pkg` (rather than just from the App Store), would it be feasible for there to be a cask formula for installing Xcode via Homebrew? I ask because I'm working on documentation for contributing to another project here on GitHub, and being able to install Xcode via the command line (especially via Homebrew) would simplify the steps for getting potential contributors up and running.

I also recently came across [xcode-install](https://github.com/xcpretty/xcode-install), which provides a command-line interface for installing Xcode, but one of the weird limitations is that you have to know in advance what version of Xcode you want to install, and the command for listing available versions requires you to log in with an Apple ID (even though the command for actually installing Xcode does not). [Someone has requested](https://github.com/xcpretty/xcode-install/issues/344) the ability to install the latest version of Xcode without specifying the version number, but after a year and a half it still has yet to be implemented. This particular ability (installing the latest non-beta version of a package) is part of how Homebrew does things...

**Comment by SMillerDev** [comment]
As far as I know you need to be logged in to the app store or the developer portal to download xcode. If there is a general unsecured download link however you could submit a cask.

If you can't you can try installing it with the `mas` formula.

**Reply by elsiehupp** [reply]
It looks like the direct links require authentication (i.e. they don't work with `curl`), so that's a non-starter. Thanks for letting me know about `mas`, though!
